src/render_tasks/d3d12_path_tracer.hpp

```diff
@@ -329,30 +329,21 @@ namespace wr
 				fg.GetPredecessorData<CubemapConvolutionTaskData>();
 
 				// Get skybox
-				if (scene_graph.m_skybox.has_value())
-				{
-					auto skybox_t = static_cast<d3d12::TextureResource*>(scene_graph.m_skybox.value().m_pool->GetTextureResource(scene_graph.m_skybox.value()));
-					d3d12::SetRTShaderSRV(cmd_list, 0, COMPILATION_EVAL(rs_layout::GetHeapLoc(params::path_tracing, params::PathTracingE::SKYBOX)), skybox_t);
-				}
-
-				// Get Pre-filtered environment
-				if (scene_graph.m_skybox.has_value())
+				if (SkyboxNode *skybox = scene_graph.GetCurrentSkybox().operator->())

```

Comment:
`.operator->()` I don't think this is necessary

Reply:
Replaced with .get() instead, intellisense didn't show it as an option.